### ISSUE TITLE
Feature: Add support for MQTT 

### ISSUE THREAD

**Issue by Timtam94** [airtai/faststream#956]
Hi there, 

in a lot of my projects in the IoT space I tend to work with RabbitMQ and Kafka but also MQTT. I would really appreciate a framework such as yours, that combine these protocols into one. 

Thank you very much for your work!

**Comment by Lancetnik** [comment]
Thank you for your kind words! We'll support for MQTT for sure in the future, but we should finish with the brokers already in backlog at first 😞

**Comment by davorrunje** [comment]
@Timtam94 what lib are you currently using for MQTT?

**Comment by Timtam94** [comment]
Thanks for your reply! I am currently using [paho-mqtt](https://pypi.org/project/paho-mqtt/)

**Comment by davorrunje** [comment]
How stable it is? It looks not well maintained

**Comment by Lancetnik** [comment]
As I know, this is the only one lib for MQTT in python
Probably, it has no commits last time cuz MQTT has no changes for decades

**Comment by Timtam94** [comment]
Indeed, I am not aware of another library than that. So far I had no issues with it. 

In the industrial sector, MQTT is one of the major protocols, despite it's age.

**Comment by gegnew** [comment]
There's also an async wrapper for `paho`: https://github.com/sbtinstruments/aiomqtt

**Comment by skewty** [comment]
> > There's also an async wrapper for `paho`: https://github.com/sbtinstruments/aiomqtt
> 
> Yep, I known about it But I am not sure about using extra dependencie... Probably, we should (and can) wrap paho by ourselves the same way

aiomqtt's dependencies are minimal. Here they are from pyproject.toml

```toml
[tool.poetry.dependencies]
python = "^3.8"
paho-mqtt = "^2.1.0"
typing-extensions = {version = "^4.4.0", markers = "python_version < '3.10'"}
```

Paho recently had a v2 release which aiomqtt is using. Paho uses a legacy callback system and is rather ugly (very non pythonic) to work with. The aiomqtt team is careful not to couple too deep into paho client as to ensure aiomqtt remains easy to maintain. The wrapper is light but non trivial. aiomqtt is likely worth coupling for this reason.

**Comment by benedikt-bartscher** [comment]
I know, it's not native mqtt, but you can already use mqtt via the [RabbitMQ MQTT Plugin](https://www.rabbitmq.com/docs/mqtt).

**Comment by vvanglro** [comment]
This looks great too. 
https://github.com/agronholm/mqttproto

**Comment by Lancetnik** [comment]
> This looks great too. https://github.com/agronholm/mqttproto

Yeah, but it is still imperative syntax

Is it really maintained by the man, created **anyio** and **aspcheduler**? I think, we can contact with him and use this project as an underline client this case

**Comment by legout** [comment]
Any updates on this? We ar running kafka, mqtt and nats in our IOT landscape. Using one library for all of them would be awesome.

**Comment by Lancetnik** [comment]
> Any updates on this? We ar running kafka, mqtt and nats in our IOT landscape. Using one library for all of them would be awesome.

I hope you can start running it at the end of this summer.

**Comment by empicano** [comment]
Shameless plug for https://github.com/empicano/mqtt5, just in case that's interesting for you guys. I'll refactor aiomqtt to build on mqtt5 instead of paho over the next weeks.

mqtt5/aiomqtt or not, let me know in case I can assist anywhere! 🙂

**Comment by Lancetnik** [comment]
It's about to get merged. Scheduled to 0.7 pretty soon
